frontend/ui/widget/textboxwidget.lua

```diff
@@ -1088,7 +1088,12 @@ function TextBoxWidget:getCharPos()
 end
 
 function TextBoxWidget:getSize()
-    return Geom:new{ w = self.width, h = self._bb:getHeight()}
+    -- Make sure we actually have a BB, in case we're recycling an instance... (c.f., #8241)
+    if not self._bb then
+        self:_renderText(self.virtual_line_num, self.virtual_line_num + self.lines_per_page - 1)
```

Comment:
That takes care of creating one all by itself properly?

Reply:
Yup!

Reply:
https://github.com/koreader/koreader/issues/8241#issuecomment-926017666

> I'd prefer a simpler self:update(), to have it look less scary :)

Reply:
And I find the extra layer of indirection this would add *more* confusing ;).

Reply:
(And the `free` call inlined in `update` may come back to bite us in the ass, too).

Reply:
Well, ok then.

(I would then probably wrap all others `self:_renderText(...)` in a `self:_update()` or something like that.)

Reply:
> (I would then probably wrap all others `self:_renderText(...)` in a `self:_update()` or something like that.)

Yeah, that sounds better ;).